source/comp/markv_autogen.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "markv_autogen.h"
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <vector>
+#include <unordered_map>
+
+#include "spirv/1.2/spirv.h"
+
+using spvutils::HuffmanCodec;
+
+namespace {
+
+// Signals that the value is not in the coding scheme and a fallback method
+// needs to be used.
+const uint64_t kMarkvNoneOfTheAbove = GetMarkvNonOfTheAbove();
+
+inline uint32_t CombineOpcodeAndNumOperands(uint32_t opcode,
+                                            uint32_t num_operands) {
+  return opcode | (num_operands << 16);
+}
+
+}  // namespace
+
+#include "markv_autogen.inc"
```

Comment:
This comment was not addressed, and it's pretty important:

You should describe the procedure for generating this .inc file, since you are checking it in.
Just need an overview of the form:

1. Gather a corpus of shaders: How to save it to disk for use.
2. Train a model by gathering statistics over a corpus. (and say how to run the stats gathering and codegen over the corpus)

Reply:
Done.